### PR TITLE
support pagination with POST

### PR DESCRIPTION
**Pull request by Azdaroth**

### CODE REVIEW

**Comment by Azdaroth** [comment]
This needs some refactoring, the code looks really ugly...

**Comment by Azdaroth** [comment]
Made some clean up, much better now, was wondering if we should split `paginate` to two methods for handling POST and GET pagination, but I think the complexity of this extraction would be not worth it and this conditional is not that bad.

File: lib/bookingsync/api/client/rentals.rb [hunk 1]
```diff
@@ -31,7 +31,10 @@ def rentals(options = {}, &block)
       def rentals_search(options = {}, &block)
         ids = Array(options.delete(:ids))
         path = ["rentals", ids.join(","), "search"].compact.join("/")
-        paginate path, options, &block
+        defaults = {
```
Comment (Fredar) [inline lib/bookingsync/api/client/rentals.rb]: Can be single line?

Comment (Azdaroth) [inline lib/bookingsync/api/client/rentals.rb]: Yeah, might look better.

File: lib/bookingsync/api/client.rb [hunk 2]
```diff
@@ -209,7 +214,7 @@ def paginate(path, options = {}, &block)
               first_request = false
             end
             break unless response.relations[:next]
-            response = response.relations[:next].get
+            response = response.relations[:next].public_send(request_method)
```
Comment (Mandaryn) [inline lib/bookingsync/api/client.rb]: why not use `call` directly?

Comment (Mandaryn) [inline lib/bookingsync/api/client.rb]: i mean to have the call method on relation like before, and use that instead of using `public_send`?

Comment (Azdaroth) [inline lib/bookingsync/api/client.rb]: Could be an option, will check that approach.

Comment (Azdaroth) [inline lib/bookingsync/api/client.rb]: Refactored a bit handling these requests for relations, looks better now I guess.

**Comment by Azdaroth** [comment]
Should not be merged until the POST pagination for Rentals#search is deployed on core to production.

File: lib/bookingsync/api/client.rb (hunk 2 above)
Comment (Mandaryn) [inline lib/bookingsync/api/client.rb]: still think it should look like
```
response = response.relations[:next].call(request_method)
```
and then this call be used internally in relations by get and post

Comment (Azdaroth) [inline lib/bookingsync/api/client.rb]: It's outdated diff, no? Currently it's `response = response.relations[:next].call({}, method: request_method)`

File: lib/bookingsync/api/client.rb [hunk 3]
```diff
@@ -209,7 +214,7 @@ def paginate(path, options = {}, &block)
               first_request = false
             end
             break unless response.relations[:next]
-            response = response.relations[:next].get
+            response = response.relations[:next].call({}, method: request_method)
```
Comment (Mandaryn) [inline lib/bookingsync/api/client.rb]: makes sense and seems not to be braking methods signatures
might blow on nils if someone forced it somewhere, but looks good

**Comment by Mandaryn** [comment]
👍

**Pull request merged**